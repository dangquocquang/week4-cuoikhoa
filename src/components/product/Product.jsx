import React, { Component } from 'react';

import store from '../../store';
import { addToCart } from '../../actions/cart-actions';

export default class Product extends Component {
  handleCartClick = () => {
    console.log('add to cart');
    store.dispatch(addToCart(this.props));
  };

  render() {
    const { name, image, image2, category, discountPrice, price } = this.props;

    return (
      <div className="col-xl-4 col-lg-6 col-md-6">
        <div className="product-wrapper mb-50">
          <div className="product-img mb-25">
            <a href="#">
              <img src={image} alt={name} />
              <img className="secondary-img" src={image2} alt={name} />
            </a>
            <div className="product-action text-center">
              <a title="Shoppingb Cart" onClick={this.handleCartClick}>
                <i className="fas fa-shopping-cart" />
              </a>
              <a href="#" title="Quick View">
                <i className="fas fa-search" />
              </a>
            </div>
          </div>
          <div className="product-content pr-0">
            <div className="pro-cat mb-10">
              <a href="#">{category}</a>
            </div>
            <h4>
              <a href="#">{name}</a>
            </h4>
            <div className="product-meta">
              <div className="pro-price">
                <span>${discountPrice} USD</span>
                <span className="old-price">${price} USD</span>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
