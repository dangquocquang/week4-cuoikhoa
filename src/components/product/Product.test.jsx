import React from 'react';
import renderer from 'react-test-renderer';
import Product from './Product.jsx';

const mockProduct = {
  id: '0002',
  name: 'Minimal Deco Furniture',
  image: './assets/pro2.jpg',
  image2: './assets/pro3.jpg',
  category: 'furniture',
  discountPrice: '119.00',
  price: '230.00',
};

test('Product component', () => {
  const component = renderer.create(<Product {...mockProduct} />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
