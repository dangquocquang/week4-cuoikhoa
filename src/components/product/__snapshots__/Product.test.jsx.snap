// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product component 1`] = `
<div
  className="col-xl-4 col-lg-6 col-md-6"
>
  <div
    className="product-wrapper mb-50"
  >
    <div
      className="product-img mb-25"
    >
      <a
        href="#"
      >
        <img
          alt="Minimal Deco Furniture"
          src="./assets/pro2.jpg"
        />
        <img
          alt="Minimal Deco Furniture"
          className="secondary-img"
          src="./assets/pro3.jpg"
        />
      </a>
      <div
        className="product-action text-center"
      >
        <a
          onClick={[Function]}
          title="Shoppingb Cart"
        >
          <i
            className="fas fa-shopping-cart"
          />
        </a>
        <a
          href="#"
          title="Quick View"
        >
          <i
            className="fas fa-search"
          />
        </a>
      </div>
    </div>
    <div
      className="product-content pr-0"
    >
      <div
        className="pro-cat mb-10"
      >
        <a
          href="#"
        >
          furniture
        </a>
      </div>
      <h4>
        <a
          href="#"
        >
          Minimal Deco Furniture
        </a>
      </h4>
      <div
        className="product-meta"
      >
        <div
          className="pro-price"
        >
          <span>
            $
            119.00
             USD
          </span>
          <span
            className="old-price"
          >
            $
            230.00
             USD
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
